pkg/ipamd/ipamd_test.go

```diff
@@ -1311,7 +1311,13 @@ func TestIPAMContext_filterUnmanagedENIs(t *testing.T) {
 			c := &IPAMContext{
 				awsClient:                mockAWSUtils,
 				enableManageUntaggedMode: true}
-			mockAWSUtils.EXPECT().SetUnmanagedENIs(tt.unmanagedenis).AnyTimes()
+
+			mockAWSUtils.EXPECT().SetUnmanagedENIs(gomock.Any()).
+				Do(func(args []string) {
+					sort.Strings(tt.unmanagedenis)
```

Comment:
This looks fine. Not sure why the unit tests are failing. Perhaps there is flake elsewhere too?

Reply:
Had failed with this diff.
```
=== RUN   TestIPAMContext_filterUnmanagedENIs/Secondary_ENI_not_CNI_created_and_Tertiary_ENI_CNI_created
--- FAIL: TestIPAMContext_filterUnmanagedENIs (0.00s)
    --- PASS: TestIPAMContext_filterUnmanagedENIs/No_tags_at_all (0.00s)
    --- PASS: TestIPAMContext_filterUnmanagedENIs/Primary_ENI_unmanaged (0.00s)
    --- PASS: TestIPAMContext_filterUnmanagedENIs/Secondary/Tertiary_ENI_unmanaged (0.00s)
    --- PASS: TestIPAMContext_filterUnmanagedENIs/Secondary_ENI_unmanaged (0.00s)
    --- PASS: TestIPAMContext_filterUnmanagedENIs/Secondary_ENI_unmanaged_and_Tertiary_ENI_CNI_created (0.00s)
    assertions.go:262: 
        	Error Trace:	ipamd_test.go:1319
        	            				value.go:556
        	            				value.go:339
        	            				call.go:152
        	            				controller.go:221
        	            				awsutils_mocks.go:469
        	            				ipamd.go:296
        	            				ipamd_test.go:1321
        	Error:      	Not equal: 
        	            	expected: []string{"eni-00000001"}
        	            	actual  : []string(nil)
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,2 @@
        	            	-([]string) (len=1) {
        	            	- (string) (len=12) "eni-00000001"
        	            	-}
        	            	+([]string) <nil>
        	            	 
        	Test:       	TestIPAMContext_filterUnmanagedENIs/Secondary_ENI_unmanaged_and_Tertiary_ENI_CNI_created
    --- FAIL: TestIPAMContext_filterUnmanagedENIs/Secondary_ENI_not_CNI_created_and_Tertiary_ENI_CNI_created (0.00s)```

Reply:
Yes, it was due to a wrong expected value in the test case for `Secondary ENI not CNI created and Tertiary ENI CNI created`. Fixed it in the latest commit in this PR.

Reply:
@vikasmb as discussed offline we need to return `nil`. This test should have failed always since expected o/p was wrong but it was successful most of the time. So even after changing this to nil, the function might return nil or eni-1 and still fail.

I think we need to fix this - https://github.com/aws/amazon-vpc-cni-k8s/blob/5f79f942dcfcc694125db998b1ffa46bea72b8ee/pkg/ipamd/ipamd_test.go#L1318-L1329